modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/ProjectedTuple.java

```diff
@@ -17,26 +17,22 @@
 
 package org.apache.ignite.internal.sql.engine.util;
 
-import java.nio.ByteBuffer;
 import org.apache.ignite.internal.binarytuple.BinaryTupleBuilder;
-import org.apache.ignite.internal.binarytuple.BinaryTupleParser;
 import org.apache.ignite.internal.binarytuple.BinaryTupleParser.Sink;
 import org.apache.ignite.internal.lang.InternalTuple;
 import org.apache.ignite.internal.schema.BinaryTuple;
+import org.apache.ignite.internal.schema.InternalTupleEx;
 
 /**
- * A projected tuple that aware of the format of delegate.
- *
- * <p>That is, the format of delegate is known to be Binary Tuple, thus it's possible to avoid unnecessary
- * (de-)serialization during tuple normalization.
+ * A projected tuple wrapper that is best effort to avoiding unnecessary (de-)serialization during tuple normalization.
  *
  * <p>It's up to the caller to get sure that provided tuple respect the format.
  *
  * <p>Not thread safe!
  *
  * @see AbstractProjectedTuple
  */
-public class FormatAwareProjectedTuple extends AbstractProjectedTuple {
+public class ProjectedTuple extends AbstractProjectedTuple {
```

Comment:
probably we should merge `ProjectedTuple` and `AbstractProjectedTuple` altogether. So we will have only `ProjectedTuple` and extension which append virtual columns. WDYT?